src/GraphQL/Types/ComplexGraphType.cs

```diff
@@ -0,0 +1,144 @@
+        }
+
+        public FieldType Field(
+            Type type,
+            string name,
+            string description = null,
+            QueryArguments arguments = null,
+            Func<ResolveFieldContext<TSourceType>, object> resolve = null,
+            string deprecationReason = null)
+        {
+
+            return AddField(new FieldType
+            {
+                Name = name,
+                Description = description,
+                DeprecationReason = deprecationReason,
+                Type = type,
+                Arguments = arguments,
+                Resolver = resolve != null 
+                    ? new FuncFieldResolver<TSourceType, object>(resolve) 
+                    : null,
+            });
+        }
+
+        public FieldType Field<TGraphType>(
+            string name,
+            string description = null,
+            QueryArguments arguments = null,
+            Func<ResolveFieldContext<TSourceType>, object> resolve = null,
+            string deprecationReason = null)
+            where TGraphType : IGraphType
+        {
+            return AddField(new FieldType
+            {
+                Name = name,
+                Description = description,
+                DeprecationReason = deprecationReason,
+                Type = typeof(TGraphType),
+                Arguments = arguments,
+                Resolver = resolve != null
+                    ? new FuncFieldResolver<TSourceType, object>(resolve)
+                    : null,
+            });
+        }
+
+        public FieldBuilder<TSourceType, TReturnType> Field<TGraphType, TReturnType>()
+        {
+            var builder = FieldBuilder.Create<TSourceType, TReturnType>(typeof(TGraphType));
+            AddField(builder.FieldType);
+            return builder;
+        }
+
+        public FieldBuilder<TSourceType, object> Field<TGraphType>()
+        {
+            var builder = FieldBuilder.Create<TSourceType, object>(typeof(TGraphType));
+            AddField(builder.FieldType);
+            return builder;
+        }
+
+        public FieldBuilder<TSourceType, TProperty> Field<TProperty>(
+           string name,
+           Expression<Func<TSourceType, TProperty>> expression,
+           bool nullable = false,
+           Type type = null)
+        {
+            type = type ?? typeof(TProperty).GetGraphTypeFromType(nullable);
+
+            var builder = FieldBuilder.Create<TSourceType, TProperty>(type)
+                .Resolve(new ExpressionFieldResolver<TSourceType, TProperty>(expression))
+                .Name(name);
+
+            AddField(builder.FieldType);
+            return builder;
+        }
+
+        public FieldBuilder<TSourceType, TProperty> Field<TProperty>(
+            Expression<Func<TSourceType, TProperty>> expression,
+            bool nullable = false,
+            Type type = null)
+        {
+            string name;
+            try {
+                name = expression.NameOf().ToCamelCase();
```

Comment:
does it make a difference in terms of final json payload? I went with camel case since it's more idiomatic json/graphql, but if that's handled by the json serializer sticking with pascal case here seems reasonable as well.